submit/submit

```diff
@@ -0,0 +1,534 @@
+
+    epilog_parts = [
+        "Explanation of submission options:",
+        contests_part_one,
+        contests_part_two,
+        problem_part,
+        '''When not specified, PROBLEM defaults to the first FILENAME excluding the
+extension. For example, 'B.java' will indicate problem 'B'.''',
+        language_part,
+        '''The default for LANGUAGE is the extension of FILENAME. For example,
+'B.java' will indicate a Java solution.''',
+        "Set URL to the base address of the webinterface without the 'api/' suffix.\n" +
+        ("The (pre)configured URL is '{}'\n".format(baseurl) if baseurl != '' else '') +
+        "Credentials are read from ~/.netrc (see netrc(4) for details).",
+        "Examples:",
+    '''Submit problem 'b' in Java:
+    {} b.java'''.format(sys.argv[0]),
+    '''Submit problem 'z' in C# for contest 'demo':
+    {} --contest=demo z.cs'''.format(sys.argv[0]),
+    '''Submit problem 'e' in C++:
+    {} --problem e --language=cpp ProblemE.cc'''.format(sys.argv[0]),
+    '''Submit problem 'hello' in C (options override the defaults from FILENAME):
+    {} -p hello -l C HelloWorld.cpp'''.format(sys.argv[0]),
+    '''Submit multiple files (the problem and language are taken from the first):
+    {} hello.java message.java'''.format(sys.argv[0]),
+    ]
+
+    return "\n\n".join(part for part in epilog_parts if part != None) 
+
+def do_api_submit():
+    '''Submit to the API with the given data'''
+
+    url = baseurl + "api/" + api_version + "contests/" + my_contest['id'] + "/submissions"
+
+    data = {
+        'problem': my_problem['id'],
+        'language': my_language['id'],
+    }
+    if entry_point is not None:
+        data['entry_point'] = entry_point
+
+    files = []
+    for filename in filenames:
+        files.append(('code[]', open(filename, 'rb')))
+
+    logging.info('connecting to {}'.format(url))
+
+    response = requests.post(url, data=data, files=files)
+
+    logging.debug("API call 'submissions' returned:\n{}".format(response.text))
+
+    # The connection worked, but we may have received an HTTP error
+    if response.status_code >= 300:
+        print(response.text)
+        if response.status_code == 401:
+            raise RuntimeError('authentication failed, please check your DOMjudge credentials.')
+        else:
+            raise RuntimeError('Submission failed (code {})'.format(response.status_code))
+
+    # We got a successful HTTP response. It worked.
+    # But check that we indeed received a submission ID.
+
+    try:
+        submission = json.loads(response.text)
+    except json.decoder.JSONDecodeError as e:
+        error('parsing REST API output: {}'.format(e))
+
+    if (not isinstance(submission, dict)
+        or not 'id' in submission
+        or not isinstance(submission['id'], str)):
+        error('REST API returned unexpected JSON data')
+
+    print('Submission received, id = s{}'.format(submission['id']))
+
+version_text = '''
```

Comment:
I don't see any version info here. Should it contain any?

Reply:
It was there before, but we dropped it to make the submit client standalone. However, we still pass it to the `--version` flag, so that is why this variable is named like this

Reply:
Do we want to give this a separate version number  then?

cc @thijskh

Reply:
We could hardcode a version number.
Or we can insert a version number/git commit at tarball build time instead of at runtime.
Or just leave it...

Reply:
I will leave it for now and let someone else decide and implement this